libs/execution_context/ebpf_program.c

```diff
@@ -884,13 +894,28 @@ _ebpf_program_get_helper_function_address(
         *address = (uint64_t)function_address;
     } else {
         ebpf_assert(program->general_helper_provider_data != NULL);
+        if (program->general_helper_provider_data == NULL) {
```

Comment:
Same as above.

Reply:
And other places in this file.

Reply:
Note: This is a common pattern in the code to both assert and check. Removed in this instance.